snp_oracle/predictionnet/utils/dataset_manager.py

```diff
@@ -0,0 +1,338 @@
+                "columns": ",".join(miner_data.columns),
+                "shape": f"{miner_data.shape[0]},{miner_data.shape[1]}",
+                "hotkey": hotkey,
+                "predictions": json.dumps(predictions) if predictions else "",
+            }
+            if metadata:
+                full_metadata.update(metadata)
+
+            # Convert to PyArrow Table with metadata
+            table = pa.Table.from_pandas(miner_data)
+            for key, value in full_metadata.items():
+                table = table.replace_schema_metadata({**table.schema.metadata, key.encode(): str(value).encode()})
+
+            # Write Parquet file with compression
+            pq.write_table(table, file_path, compression="snappy", use_dictionary=True, use_byte_stream_split=True)
+
+            return True, {
+                "local_path": str(file_path),
+                "rows": miner_data.shape[0],
+                "columns": miner_data.shape[1],
+            }
+
+        except Exception as e:
+            bt.logging.error(f"Error in store_local_data: {str(e)}")
+            return False, {"error": str(e)}
+
+    def _configure_git_repo(self, repo: Repo):
+        """Configure Git repository with user information"""
+        git_name = os.getenv("GIT_NAME", "DatasetManager")
+        git_email = os.getenv("GIT_EMAIL", "noreply@example.com")
+
+        with repo.config_writer() as git_config:
+            git_config.set_value("user", "name", git_name)
+            git_config.set_value("user", "email", git_email)
+
+    def _setup_git_repo(self, repo_name: str) -> Tuple[Repo, Path]:
+        """Set up Git repository for batch upload"""
+        temp_dir = Path(tempfile.mkdtemp())
+        bt.logging.info(f"Created temporary directory: {temp_dir}")
+        repo_url = self.git_url_template.format(repo_name=repo_name)
+        repo_id = f"{self.organization}/{repo_name}"
+
+        try:
+            # First check if repo exists
+            try:
+                self.api.repo_info(repo_id=repo_id, repo_type="dataset")
+            except Exception:
+                # Repository doesn't exist, create it via API
+                self.api.create_repo(repo_id=repo_id, repo_type="dataset", private=False)
+                bt.logging.info(f"Created new repository: {repo_name}")
+
+            # Now clone the repository (it should exist)
+            repo = Repo.clone_from(repo_url, temp_dir)
+            bt.logging.success(f"Cloned repository: {repo_name}")
+
+            # Configure Git if needed
+            self._configure_git_repo(repo)
+
+            return repo, temp_dir
+
+        except Exception as e:
+            # Clean up temp dir if anything fails
+            if temp_dir.exists():
+                shutil.rmtree(temp_dir)
+                bt.logging.info(f"Deleted temporary directory: {temp_dir}")
+            raise Exception(f"Failed to setup repository: {str(e)}")
+
+    def batch_upload_daily_data(self) -> Tuple[bool, Dict]:
```

Comment:
The data is cached locally and then loaded at the end of the day. The reason was initially to limit hf api requests, but then I had to pivot to use git directly. This is an artifact of that choice, but I am not sure I see a reason to change it.

Reply:
Validators check that they can successfully decrypt the data file, and if this fails then the miner receives a score of zero. Will the validator be able to find these files?

Reply:
I'm not sure I follow. Are you asking if they will be able to find the unsuccessfully decrypted files? If so, the answer would be no. I don't think we want to upload files that cannot be properly decrypted.

Reply:
I'm trying to understand if the validator has / needs access to this local data cache to properly evaluate

Reply:
The validator writes the files to local storage following decryption. If the data cannot be decrypted, it will not be stored locally and therefore not be uploaded at the end of the day by the validator. The validator owns the local storage. Hayden and I had discussed whether every validator needs to store/upload this decrypted data, and the answer may be no (Do we need duplicity of decrypted miner data across validators, or just our vali for now?). The only part that matters for this portion of the integration is that the data can be successfully decrypted.